src/ResolveKeysCapableTrait.php

```diff
@@ -6,15 +6,21 @@
 
 use Psr\Container\ContainerInterface;
 
+/**
+ * Functionality for resolving a service key.
+ *
+ * @psalm-type ServiceRef = string|callable(ContainerInterface): mixed
```

Comment:
```suggestion
 * @psalm-type ServiceFactory = callable(ContainerInterface): mixed
 * @psalm-type ServiceRef = string|ServiceFactory
```

This way there's a named reference to a commonly used callable signature. Other parts of the application can reference it with `@psalm-import-type`.

Reply:
I'm not 100% on whether this is the right place to declare those types, though. In fact, looks like there might be [issues](https://github.com/vimeo/psalm/issues/7971) with importing Psalm types defined on a trait. So I'm open to discussion regarding possible better placement.

Reply:
I added the `ServiceFactory` type and moved it to the doc block of the `Service` class after confirming that Psalm was not recognizing the type in the trait's doc block.